src/gui/base/RichTextToolbar.ts

```diff
@@ -1,190 +1,195 @@
-import m, {Children, Component, VnodeDOM} from "mithril"
+import m, {Children, Component, Vnode, VnodeDOM} from "mithril"
 import {Icons} from "./icons/Icons"
-import type {Editor} from "../editor/Editor"
+import type {Editor, Listing, Style} from "../editor/Editor"
+import {Alignment} from "../editor/Editor"
 import {noOp, numberRange} from "@tutao/tutanota-utils"
-import type {ButtonAttrs} from "./ButtonN"
-import {ButtonColor, ButtonN, ButtonType} from "./ButtonN"
+import type {ButtonAttrs} from "./Button.js"
+import {Button, ButtonColor, ButtonType} from "./Button.js"
 import {size} from "../size"
-import {attachDropdown, DropdownButtonAttrs} from "./DropdownN"
-import {lang} from "../../misc/LanguageViewModel"
+import {attachDropdown} from "./Dropdown.js"
+import {lang, TranslationKey} from "../../misc/LanguageViewModel"
 import {animations, height, opacity} from "../animation/Animations"
 import {client} from "../../misc/ClientDetector"
 import {BrowserType} from "../../misc/ClientConstants"
 
-export type Options = {
+export interface RichTextToolbarAttrs {
+	editor: Editor
 	imageButtonClickHandler?: ((ev: Event, editor: Editor) => unknown) | null
 	alignmentEnabled?: boolean
 	fontSizeEnabled?: boolean
 	customButtonAttrs?: Array<ButtonAttrs>
 }
 
-export class RichTextToolbar implements Component<Options> {
-	view: Component["view"]
-	selectedSize: number
+export class RichTextToolbar implements Component<RichTextToolbarAttrs> {
+	selectedSize = size.font_size_base
 
-	constructor(editor: Editor, options?: Options) {
-		this.selectedSize = size.font_size_base
-		const styleToggleAttrs: Array<ButtonAttrs> = ([
-			{
-				style: "b",
-				title: () => lang.get("formatTextBold_msg") + " (Ctrl + B)",
-				icon: Icons.Bold,
-			},
-			{
-				style: "i",
-				title: () => lang.get("formatTextItalic_msg") + " (Ctrl + I)",
-				icon: Icons.Italic,
-			},
-			{
-				style: "u",
-				title: () => lang.get("formatTextUnderline_msg") + " (Ctrl + U)",
-				icon: Icons.Underline,
-			},
-			{
-				style: "c",
-				title: "formatTextMonospace_msg",
-				icon: Icons.Code,
-			},
+	oncreate(vnode: VnodeDOM<any>): void {
+		const dom = vnode.dom as HTMLElement
+		dom.style.height = "0"
+		animateToolbar(dom, true)
+	}
+
+	onbeforeremove(vnode: VnodeDOM<any>): Promise<void> {
+		return animateToolbar(vnode.dom as HTMLElement, false)
+	}
+
+	view({attrs}: Vnode<RichTextToolbarAttrs>): Children {
+
+		try {
```

Comment:
hm probably parsing it each `view()` is not a great idea, probably the easiest is to memoize it?